src/components/uploads/api.js

```diff
@@ -79,18 +88,17 @@ export const startUpload = (
             completedCB(upload.id);
         })
         .catch((e) => {
-            if (e.details) {
-                console.error(
-                    `${e.details.code} ${JSON.stringify(e.details.context)}`
-                );
-            } else {
-                console.error(e.message);
+            let errorMessage = e;
+            if (e.data) {
+                const { error_code, ...rest } = e.data;
+                errorMessage = `${error_code} ${JSON.stringify(rest)}`;
```

Comment:
I wonder if we should have a friendlier error message for common, known errors, like `ERR_EXISTS`, and leave this generic message only for unexpected errors.

For example: `A file with that name already exists at that location`.
The full path could also be parsed from `e.data` and included, in case the user is uploading multiple files to different destination folders 🤔

Reply:
Agree with @psarando . We already do something like that in Rename Dialog https://github.com/cyverse-de/sonora/blob/d37ddaedd4f96a312018dbe6e464e1d443903ff8/src/components/data/RenameDialog.js#L56 and CreateFolderDialog https://github.com/cyverse-de/sonora/blob/d37ddaedd4f96a312018dbe6e464e1d443903ff8/src/components/data/CreateFolderDialog.js#L51